apps/api/src/domain/models/audio-item/entities/transcript.entity.ts

```diff
@@ -96,4 +100,47 @@ export class Transcript<T extends string = string> extends BaseDomainModel {
     findParticipantByName(name: string): Maybe<TranscriptParticipant> {
         return this.participants.find((participant) => participant.name === name) || NotFound;
     }
+
+    addLineItem(item: TranscriptItem): ResultOrError<this> {
+        const lineItemErrors = this.validateLineItem(item);
+
+        if (lineItemErrors.length > 0)
+            return new CannotAddInconsistentLineItemError(item, lineItemErrors);
+
+        // Avoid shared references to individual items by cloning
+        const newItems = this.items.concat(item.clone());
+
+        return this.clone({
```

Comment:
Another possibility is that all of these issues become invariant validation rules. The `addLineItem` logic then remains clear of validation checks and we defer to the `safeClone` which passes through invariant validation layer "on the way back".